src/mlpack/core/data/load_csv.hpp

```diff
@@ -94,45 +153,58 @@ class LoadCSV
     std::string line;
     while (std::getline(inFile, line))
     {
-      ++rows;
+       ++rows;
+    }
+
+    // Reset the DatasetInfo object, if needed.
+    if (info.Dimensionality() == 0)
+    {
+      info.SetDimensionality(rows);
+    }
+    else if (info.Dimensionality() != rows)
+    {
+      std::ostringstream oss;
+      oss << "data::LoadCSV(): given DatasetInfo has dimensionality "
+          << info.Dimensionality() << ", but data has dimensionality "
+          << rows;
+      throw std::invalid_argument(oss.str());
     }
-    info = DatasetMapper<MapPolicy>(rows);
 
     // Now, jump back to the beginning of the file.
     inFile.clear();
     inFile.seekg(0, std::ios::beg);
     rows = 0;
-
+    
     while (std::getline(inFile, line))
     {
       ++rows;
-      // Remove whitespace from either side.
-      boost::trim(line);
 
       if (rows == 1)
       {
         // Extract the number of columns.
-        auto findColSize = [&cols](iter_type) { ++cols; };
-        qi::parse(line.begin(), line.end(),
-            stringRule[findColSize] % delimiterRule);
+	std::pair<int, int> dimen = GetMatSize(inFile);
+        cols = dimen.second;
       }
 
       // I guess this is technically a second pass, but that's ok... still the
       // same idea...
       if (MapPolicy::NeedsFirstPass)
       {
         // In this case we must pass everything we parse to the MapPolicy.
-        auto firstPassMap = [&](const iter_type& iter)
-        {
-          std::string str(iter.begin(), iter.end());
-          boost::trim(str);
-
-          info.template MapFirstPass<T>(std::move(str), rows - 1);
-        };
+	std::string str(line.begin(), line.end());
+        
+	for(int i = 0; i < str.size(); i++)
+	{
+	  // Maybe there is a faster way to parser each element of the string
+	  // Also for now it is being considered that delimiter will always
+	  // be comma(,)
+	  if(str[i] != ',')
+	  {
+	    std::string cc(1, str[i]);
+	    info.template MapFirstPass<T>(std::move(cc), rows - 1);
+	  }
+	}
```

Comment:
Would you explain what this part of code is doing?

Reply:
str contains a whole row in this form: `1,4,7,9,T`
We are extracting each element from the str
and sending it to `MapFirstPass`

`MapFirstPass` in turn determines if the dimension is numeric or categorical.

Also, I think we can also have str as `1/t4/t7/t9/tT` so we need to handle that case as well.
Basically what delimiter rule was doing eairlier.

Reply:
I am not sure if this is the best way to do it, How do you distinguish if a string has a comma already in it ? For example, suppose if the col is as follows:
`1, "this is a string, with a comma", 2, 3`
As you can notice, in this col there are three rows, and one string has a comma inside.
You can take a look at `load_save_test.cpp` you need to be sure that the parsing is taking in consideration all these cases.
Also in mlpack matrices are always transposed :+1: